src/Datadog.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import {
+  BatchSize,
+  DatadogProvider,
+  DatadogProviderConfiguration, DdLogs,
+  SdkVerbosity,
+  UploadFrequency,
+} from '@datadog/mobile-react-native'
+import Config from 'react-native-config';
+import { DdRumReactNavigationTracking } from '@datadog/mobile-react-navigation'
+
+// Enable Datadog only in production, change to true to test locally
+const enabled = !__DEV__
+
+const clientToken = Config.DATADOG_CLIENT_TOKEN;
+if (enabled && !clientToken) {
+  throw new Error('DATADOG_CLIENT_TOKEN is required');
+}
+
+const applicationId = Config.DATADOG_APPLICATION_ID;
+if (enabled && !applicationId) {
+  throw new Error('DATADOG_APPLICATION_ID is required');
+}
+
+const datadogConfig = new DatadogProviderConfiguration(
+  clientToken,
+  __DEV__ ? 'dev' : 'prod',
+  applicationId,
+  true, // track User interactions (e.g.: Tap on buttons. You can use 'accessibilityLabel' element property to give tap action the name, otherwise element type will be reported)
+  true, // track XHR Resources
+  true, // track Errors
+);
+// Optional: Select your Datadog website (one of "US1", "EU1", "US3", "US5", "AP1" or "GOV")
+datadogConfig.site = 'US1';
+// Optional: Enable JavaScript long task collection
+datadogConfig.longTaskThresholdMs = 100;
+// Optional: enable or disable native crash reports
+datadogConfig.nativeCrashReportEnabled = true;
+// Optional: Sample RUM sessions (% of session are sent to Datadog. Default is 100%).
+datadogConfig.sessionSamplingRate = __DEV__ ? 100 : 50;
+// Optional: Sample tracing integrations for network calls between your app and your backend (% of calls to your instrumented backend are linked from the RUM view to the APM view. Default is 20%)
+// You need to specify the hosts of your backends to enable tracing with these backends
+datadogConfig.resourceTracingSamplingRate = __DEV__ ? 100 : 50;
```

Comment:
50 maybe a little bit too much we are not going to be overwhelmed you think?

Reply:
I did not know what to put, actually. Ideally, I want 100% 😄 but they cost $ 1.50 per 1000 sessions. Let's start with 20% maybe